cmake/iotjs.cmake

```diff
@@ -25,27 +25,21 @@ function(find_value RESULT VALUE VALUE_TRUE VALUE_FALSE)
   endif()
 endfunction(find_value)
 
-# System Configuration
+# System Configuration (not module)
 set(IOTJS_PLATFORM_SRC)
-if("${CMAKE_SYSTEM_NAME}" MATCHES "Linux|Tizen")
-  file(GLOB IOTJS_PLATFORM_LINUX_SRC
-       ${IOTJS_SOURCE_DIR}/platform/iotjs_*-linux.c)
-  list(APPEND IOTJS_PLATFORM_SRC ${IOTJS_PLATFORM_LINUX_SRC})
-  # in case of Tizen force the platform to Linux
-  string(REGEX REPLACE "tizen$" "linux"
-         PLATFORM_DESCRIPTOR "${PLATFORM_DESCRIPTOR}")
-endif()
+string(TOLOWER ${CMAKE_SYSTEM_NAME} IOTJS_SYSTEM_OS)
+set(PLATFORM_OS_SRC
+    ${IOTJS_SOURCE_DIR}/platform/${IOTJS_SYSTEM_OS})
+set(OS_SRC ${PLATFORM_OS_SRC}/iotjs_[^module]*${IOTJS_SYSTEM_OS}.c)
```

Comment:
I'm surprised to see this glob working as expected. I thought that `[^module]*` would reject anything that has 'm', 'o', 'd', 'u', 'l', 'e' characters in it (i.e., that it did not look for the whole string but for any of the individual letters. (I see that something like this was there in the original code - that's why I'm surprised.)

The other thing is: however we formulate the glob, don't we want to enforce a dash before the OS suffix (i.e., `-${IOTJS_SYSTEM_OS}`)?